### PR TITLE
SurfaceFluxes.jl plugin

### PR DESCRIPTION
**Pull request by LenkaNovak**
Provides a coupled test case that uses the dry Nishizawa & Kitamura (2018) formulation using `SurfaceFluxes.jl`. 

## TODO:
- [x] integrate to run with ClimaCore/ ClimaAtmos interface, ensuring type compatibility
- [x] note changes in SF and NS + revamp the SurfaceFluxes.jl repo with original developers
- [x] test for longer simtime and compare with the bulk formula test + conservation (conservation achieved)
- [x] note if still conservation problems due to divergence operator - resolved (needed to apply boundary conditions only to diffusive fluxes, not advective fluxes as was the default)
- [x] clean up @shows + document the SF functions (+ make README consistent)
- [x] hook up to CI 

Rebased from [this](https://github.com/CliMA/CouplerMachine/pull/19) PR.

### CODE REVIEW

File: experiments/ClimaCore/Project.toml [hunk 1]
```diff
@@ -1,6 +1,5 @@
 [deps]
 CLIMAParameters = "6eacf6c3-8458-43b9-ae03-caf5306d3d53"
-ClimaAtmos = "b2c96348-7fb7-4fe0-8da9-78d88439e717"
```
Comment (jb-mackay) [inline experiments/ClimaCore/Project.toml:3]: Was there a reason for these to be removed? I think we can revert these changes to this file.

File: experiments/ClimaCore/tc4_atm-lnd-sfc-fluxes/Project.toml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+[deps]
```
Comment (jb-mackay) [inline experiments/ClimaCore/tc4_atm-lnd-sfc-fluxes/Project.toml:1]: Since this shares most dependencies with the other ClimaCore examples, I think we can just add `DocStringExtensions`, `KernelAbstactions`, and `SurfaceFluxes` to the `examples/ClimaCore/Project.toml`. I guess I am also okay with each test case folder having its own `Project.toml`, but we should be consistent here.

File: Project.toml [hunk 3]
```diff
@@ -5,6 +5,7 @@ version = "0.1.0"
 
 [deps]
 Dates = "ade2ca70-3891-5945-98fb-dc099432e06a"
+Pkg = "44cfe95a-1eb2-52ea-b672-e2afdf69b78f"
```
Comment (jb-mackay) [inline Project.toml:8]: I think it is standard now to have a `Project.toml` for the test directory that is separate from the package dependencies, so we can create that now :)

File: test/experiments_ClimaCore.jl [hunk 4]
```diff
@@ -0,0 +1,80 @@
+# Experiments
+
+using Test
+using Pkg
+
+Pkg.activate(joinpath(@__DIR__,"../experiments/ClimaCore/tc4_atm-lnd-sfc-fluxes/"))
+
+Pkg.add(url="https://github.com/CliMA/ClimaAtmos.jl",rev="main" )
+Pkg.add(url="https://github.com/CliMA/SurfaceFluxes.jl",rev="sf_rc0.2" )
```
Comment (jb-mackay) [inline test/experiments_ClimaCore.jl:9]: We can add these to the `Project.toml`; we have the `Manifest.toml` checked in so the appropriate branches of these packages will be used.

File: test/experiments_ClimaCore.jl [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+Pkg.instantiate()
+Pkg.precompile()
```
Comment (jb-mackay) [inline test/experiments_ClimaCore.jl:12]: I need to look at CI setup to see if we can avoid this kind of environment activation. We could always add all this to the `test/Project.toml` I mentioned above.

File: experiments/ClimaCore/tc4_atm-lnd-sfc-fluxes/Project.toml (hunk 2 above)
Comment (jb-mackay) [reply experiments/ClimaCore/tc4_atm-lnd-sfc-fluxes/Project.toml:1]: Changing my mind here; I think it will be less convoluted to have separate projects for each experiment. This will give us greater control over dependency versioning and branches. It also follows ClimaCore.

Comment (jb-mackay) [reply experiments/ClimaCore/tc4_atm-lnd-sfc-fluxes/Project.toml:1]: We can add these as we update the other experiments in future PRs.

File: test/experiments_ClimaCore.jl [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+using Test
+
+include(joinpath(@__DIR__,"../experiments/ClimaCore/tc4_atm-lnd-sfc-fluxes/experiment.jl"))
+
+
+FT = Float64
+parameters = (
+        # timestepping parameters 
+        Δt_min = 0.01, # minimum model timestep [s]
+        timerange = (0.0, 1.0),  # start time and end time [s]
```
Comment (jb-mackay) [inline test/experiments_ClimaCore.jl:14]: I shortened this time so the test completes more quickly; @LenkaNovak is there a minimum time we want to run this for in the CI?

Comment (LenkaNovak) [reply test/experiments_ClimaCore.jl:14]: Sounds good. Since this is only an integration test, 100 time steps should be enough :).

**Comment by jb-mackay** [comment]
@charleskawczynski can I get your insight on why the Julia formatter is struggling here?

**Comment by charleskawczynski** [comment]
Looking...

**Comment by charleskawczynski** [comment]
These struct definitions are incomplete:
```julia
struct Gravity{𝒯} <: AbstractGravity{𝒯}
struct Buoyancy{𝒯} <: AbstractGravity{𝒯}
```

**Comment by charleskawczynski** [comment]
One nice trick with debugging formatter problems is opening the Julia REPL and trying to load the offending file. This often gives more informative error messages (since the formatter doesn't try to reason about the code in the same way that happens during code loading).

**Comment by jb-mackay** [comment]
@charleskawczynski does the JuliaFormatter action fail if the formatter hasn't been applied and then the formatted code committed? Or is this action failing for another reason?

**Review by jb-mackay — APPROVED** [review]
Looks good - just need the formatter to be happy :)

**Comment by charleskawczynski** [comment]
> Looks good - just need the formatter to be happy :)

We're not currently depending on the success of the formatter ([here](https://github.com/CliMA/CouplerMachine/blob/main/bors.toml#L2-L5)), so we can merge this as is. If you'd like to apply the formatter for this PR, we could do that but, since the PR is already pretty large, the formatting could be applied in a separate PR (and then add the format entry in the bors toml file).

**Comment by charleskawczynski** [comment]
bors ping

**Comment by jb-mackay** [comment]
bors ping
